SwiftAlgorithim.playground/Pages/RotationalCipher.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+struct RotationalCipher {
+    let backing:String
+    let alphabet = ["a","b","c","d","e","f",
+                    "g","h","i","j","k","l","m",
+                    "n","o","p","q","r","s","t",
+                    "u","v","w","x","y","z"]
+    func cipher(withRotationFactor: Int) ->  String {
+        let upperAlphabetIndex = 26
+        var newString = String()
+
+        for element in backing {
+            print(":")
+            if element.isLetter {
+                let elemetWasUppercase = element.isUppercase
+                let letter = String(element.lowercased())
+                if let startingIndex = alphabet.firstIndex(of: letter) {
```

Comment:
Improvements:
Instead of Array If hashMap can be used?

Reply:
Yes, That can reduce the unnecessary iterating  for index every time,  if we use the dictionary
["a": 1, "b": 2, "c": 3, "d": 4, "e": 5 ...... ]

Reply:
Actually, I was thinking of using hashmap but it also ends up with the same or maybe more no of iteration when I am searching for char on the rotational index. 

```
extension Dictionary where Value: Equatable {
    func someKey(forValue val: Value) -> Key? {
        return first(where: { $1 == val })?.key
    }
}

```

Reply:
if you are using HasMap, we would be getting the index by passing the char like

let charHash: [Charcater: Int] = ["a": 0, "b:" 1]
let currChar = ....
if let charIndex =  charHash[currChar] {  // Here we are accessing the index passing the char as key, 
                                                                     // which will use the hash value of char to find the index in the hash table
                                                                     // which has time complexity of O(1)
}

Reply:
yea makes sense. 
I think I didn't get the first comment completely but it makes sense to me now. If I understood correctly we have one hashmap which is used to access the index and the rest is the same. 

I will push the changes, in the meanwhile if you think if we are still not on the same page please let me know.
Thanks.